securesystemslib/schema.py

```diff
@@ -206,6 +206,47 @@ def check_match(self, object):
 
 
 
+class AnyNonemptyString(Schema):
+  """
+  <Purpose>
+    Matches any string with one or more characters.
+    This schema can be viewed as the Any() schema applied to Strings, but an
+    additional check is performed to ensure only strings are considered and
+    that said strings have at least one character.
+
+    Supported methods include:
+      matches(): returns a Boolean result.
+      check_match(): raises 'exceptions.FormatError' on a mismatch.
+
+  <Example Use>
+
+    >>> schema = AnyNonemptyString()
+    >>> schema.matches('')
+    False
+    >>> schema.matches('a string')
+    True
+    >>> schema.matches(['a'])
+    False
+    >>> schema.matches(3)
+    False
+    >>> schema.matches(u'a unicode string')
+    True
+    >>> schema.matches({})
+    False
+  """
+
+  def __init__(self):
+    pass
+
+
+  def check_match(self, object):
+    if not isinstance(object, six.string_types):
+      raise securesystemslib.exceptions.FormatError('Expected a string'
+          ' but got ' + repr(object))
```

Comment:
Minor suggestion: We could subclass from AnyString and replace this first check with `super().check_match`.
Subclassing would also make it safe to remove the constructor here. Because the implicitly called `AnyString.__init__(): pass` will then prevent us from calling any further upper-level constructors (... which for some reason seem to be desired, or otherwise I don't understand the many `__init__(): pass` here).

Reply:
Good call! `super()` is a bit messy since this code (appears to be) Python 2.7 compatible, so I called `AnyString.check_match(self, object)` explicitly. We could use `super` if we turned the top-level `Schema` class into a "new-style" one, but I figured that this was less invasive for a small change.